plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java

```diff
@@ -48,6 +48,8 @@
     private static final String[] QUERY_KEYWORDS = new String[] { SQLConstants.KEYWORD_SELECT };
     private static final String[] EXEC_KEYWORDS = new String[0];
     private static final String[] DDL_KEYWORDS = new String[0];
+    private static final Collection<String> TRANSACTION_NON_MODIFYING_KEYWORDS =
```

Comment:
Can we have a plain array here? There's not much entries to search from, so performance is not so crucial here, I believe. Also, it's an additional overhead.